### PR TITLE
Add JPO to Start framework - take 2

### PR DESCRIPTION
**Pull request by rcoll**
### This is still a work-in-progress.

The purpose of this PR is to add the Jetpack Onboarding process to Calypso. Users from specific hosts (or possibly under other conditions) are given a special connect URL that includes `onboarding=1`. Once the user has connected their Jetpack site, they will be automatically redirected directly to the JPO flow. It's still unclear where the user ends up after the flow is complete (TBD).

### Testing Instructions:
Boot up your Calypso dev on this branch.

Apply [this branch](https://github.com/Automattic/jetpack/pull/7468) to the Jetpack plugin on a Jetpack site. Disconnect and then reconnect that site to Jetpack. Grab the connection URL (`/jetpack/connect/authorize/.....`) during the process and change `https://wordpress.com` to `http://calypso.localhost:3000` in the URL.

Paste the connection URL in your browser and follow the normal process to connect Jetpack. You should then be taken to the onboarding flow.

### CODE REVIEW

File: client/signup/steps/jpo-site-type/select-business-address.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+				<Card>
+					<FormLabel>{ translate( 'Business Name' ) }</FormLabel>
+					<FormTextInput />
+					<FormLabel>{ translate( 'Street Address' ) }</FormLabel>
+					<FormTextInput />
+					<FormLabel>{ translate( 'City' ) }</FormLabel>
+					<FormTextInput />
+					<FormLabel>{ translate( 'State' ) }</FormLabel>
+					<FormTextInput />
+					<FormLabel>{ translate( 'ZIP Code' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-site-type/select-business-address.jsx]: Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Zip Code' )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jpo-steps)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-site-type/select-business-address.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		if ( ! this.props.current ) {
+			return ( <div /> );
+		}
+
+		return ( 
+			<div className="jpo__site-type-wrapper business-address">
+				<Card>
+					<FormLabel>{ translate( 'Business Name' ) }</FormLabel>
+					<FormTextInput />
+					<FormLabel>{ translate( 'Street Address' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-site-type/select-business-address.jsx]: Hi! I've found a possible matching string that has already been translated **32** times:
`translate( 'Invalid street address' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jpo-steps)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-site-type/select-business-personal.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		return ( 
+			<div className="jpo__site-type-wrapper">
+				<div className="jpo__site-type-row jpo__site-type-row-small">
+					<Card>
+						<PersonalGraphic />
+						<Button onClick={ this.props.onSelectPersonal }>{ translate( 'Personal site' ) }</Button>
+					</Card>
+					<Card>
+						<BusinessGraphic />
+						<Button onClick={ this.props.onSelectBusiness }>{ translate( 'Business site' ) }</Button>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-site-type/select-business-personal.jsx]: Hi! I've found a possible matching string that has already been translated **25** times:
`translate( 'Business Site' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-site-type/select-business-personal.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		if ( ! this.props.current ) {
+			return ( <div /> );
+		}
+
+		return ( 
+			<div className="jpo__site-type-wrapper">
+				<div className="jpo__site-type-row jpo__site-type-row-small">
+					<Card>
+						<PersonalGraphic />
+						<Button onClick={ this.props.onSelectPersonal }>{ translate( 'Personal site' ) }</Button>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-site-type/select-business-personal.jsx]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Personal Site' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-contact-form/index.jsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+		this.props.goToNextStep();
+	},
+
+	renderStepContent() {
+		return (
+			<div className="jpo__contact-form-wrapper">
+				<Card>
+					<ContactUsGraphic />
+					<Button>Add a contact form</Button>
+					<div className="jpo__contact-form-description">Not sure? You can skip this step and add a contact form later.</div>
```
Comment (eliorivero) [inline client/signup/steps/jpo-contact-form/index.jsx]: This and the previous line need `translate`

File: client/signup/steps/jpo-site-type/select-business-address.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+				<Card>
+					<FormLabel>{ translate( 'Business Name' ) }</FormLabel>
+					<FormTextInput onChange={ this.props.onInputBusinessName } />
+					<FormLabel>{ translate( 'Street Address' ) }</FormLabel>
+					<FormTextInput onChange={ this.props.onInputStreetAddress } />
+					<FormLabel>{ translate( 'City' ) }</FormLabel>
+					<FormTextInput onChange={ this.props.onInputCity } />
+					<FormLabel>{ translate( 'State' ) }</FormLabel>
+					<FormTextInput onChange={ this.props.onInputState } />
+					<FormLabel>{ translate( 'ZIP Code' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-site-type/select-business-address.jsx]: Hi! I've found a possible matching string that has already been translated **31** times:
`translate( 'Zip Code' )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jpo-steps)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-site-type/select-business-address.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+		if ( ! this.props.current ) {
+			return ( <div /> );
+		}
+
+		return ( 
+			<div className="jpo__site-type-wrapper business-address">
+				<Card>
+					<FormLabel>{ translate( 'Business Name' ) }</FormLabel>
+					<FormTextInput onChange={ this.props.onInputBusinessName } />
+					<FormLabel>{ translate( 'Street Address' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-site-type/select-business-address.jsx]: Hi! I've found a possible matching string that has already been translated **32** times:
`translate( 'Invalid street address' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jpo-steps)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-homepage/index.jsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+			<div className="jpo__homepage-wrapper">
+				<div className="jpo__homepage-row">
+					<Card>
+						<NewsGraphic />
+						<Button onClick={ this.onSelectNews }>{ translate( 'Recent news or updates' ) }</Button>
+						<div className="jpo__homepage-description">{ translate( 'We can pull the latest information into your homepage for you.' ) }</div>
+					</Card>
+					<Card>
+						<StaticGraphic />
+						<Button onClick={ this.onSelectStatic }>{ translate( 'A static welcome page' ) }</Button>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-homepage/index.jsx]: Hi! I've found a possible matching string that has already been translated **93** times:
`translate( 'A static page' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-summary/index.jsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+			</div>
+		);
+	},
+
+	render() {
+		let siteTitle = ( 'undefined' !== typeof this.props.signupProgress[0].jpoSiteTitle )
+			? this.props.signupProgress[0].jpoSiteTitle.siteTitle
+			: translate( 'your new site' );
+
+		const headerText = translate( 'Congratulations! ' ) + siteTitle + translate( ' is on its way.' );
```
Comment (a8ci18n) [inline client/signup/steps/jpo-summary/index.jsx]: Hi! I've found a possible matching string that has already been translated **70** times:
`translate( 'Congratulations!' )` *ES Score: 33*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-summary/index.jsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+							<td>
+								<div className="jpo__summary-col-header">Configure more of your site:</div>
+								<ul>
+									{ 'undefined' === typeof this.props.signupProgress[0].jpoSiteTitle ? <li onClick={ this.goToStepOne }>{ translate( 'Site Title & Description' ) }</li> : null }
+									{ 'undefined' === typeof this.props.signupProgress[1].jpoSiteType ? <li onClick={ this.goToStepTwo }>{ translate( 'Type of Site' ) }</li> : null }
+									{ 'undefined' === typeof this.props.signupProgress[2].jpoHomepage ? <li onClick={ this.goToStepThree }>{ translate( 'Type of Homepage' ) }</li> : null }
+									{ 'undefined' === typeof this.props.signupProgress[3].jpoContactForm ? <li onClick={ this.goToStepFour }>{ translate( 'Contact Us Form' ) }</li> : null }
+									{ ! connectionToJetpackComplete ? <li>{ translate( 'Connection to Jetpack' ) }</li> : null }
+									<li>{ translate( 'Choose a Theme' ) }</li>
+									<li>{ translate( 'Add a Site Address' ) }</li>
```
Comment (a8ci18n) [inline client/signup/steps/jpo-summary/index.jsx]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Site Address', 'NULL', { context: 'NULL'} )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jpo-steps)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-summary/index.jsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+					<tbody>
+						<tr>
+							<td>
+								<div className="jpo__summary-col-header">Steps you've completed:</div>
+								<ul>
+									{ 'undefined' !== typeof this.props.signupProgress[0].jpoSiteTitle ? <li>{ checkMark } { translate( 'Site Title & Description' ) }</li> : null }
+									{ 'undefined' !== typeof this.props.signupProgress[1].jpoSiteType ? <li>{ checkMark } { translate( 'Type of Site' ) }</li> : null }
+									{ 'undefined' !== typeof this.props.signupProgress[2].jpoHomepage ? <li>{ checkMark } { translate( 'Type of Homepage' ) }</li> : null }
+									{ 'undefined' !== typeof this.props.signupProgress[3].jpoContactForm ? <li>{ checkMark } { translate( 'Contact Us Form' ) }</li> : null }
+									{ connectionToJetpackComplete ? <li>{ checkMark } { translate( 'Connection to Jetpack' ) }</li> : null }
```
Comment (a8ci18n) [inline client/signup/steps/jpo-summary/index.jsx]: Hi! I've found a possible matching string that has already been translated **25** times:
`translate( 'Jetpack Connection' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/jpo-summary/index.jsx [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+				<table>
+					<tbody>
+						<tr>
+							<td>
+								<div className="jpo__summary-col-header">Steps you've completed:</div>
+								<ul>
+									{ 'undefined' !== typeof this.props.signupProgress[0].jpoSiteTitle ? <li>{ checkMark } { translate( 'Site Title & Description' ) }</li> : null }
+									{ 'undefined' !== typeof this.props.signupProgress[1].jpoSiteType ? <li>{ checkMark } { translate( 'Type of Site' ) }</li> : null }
+									{ 'undefined' !== typeof this.props.signupProgress[2].jpoHomepage ? <li>{ checkMark } { translate( 'Type of Homepage' ) }</li> : null }
+									{ 'undefined' !== typeof this.props.signupProgress[3].jpoContactForm ? <li>{ checkMark } { translate( 'Contact Us Form' ) }</li> : null }
```
Comment (a8ci18n) [inline client/signup/steps/jpo-summary/index.jsx]: Hi! I've found a possible matching string that has already been translated **40** times:
`translate( 'Contact Us' )` *ES Score: 7*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jpo-steps)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request closed without merging**